### PR TITLE
Feature/heartbeat

### PR DESCRIPTION
**Pull request by andrewmil**
Removed old functionality from console in views.py. Removed .html files. 

Added a basic heartbeat along with unit testing functionality

### CODE REVIEW

File: CHANGELOG.md [hunk 1]
```diff
@@ -7,5 +7,10 @@
   - Add QBS template
   - Add all environment variables to README
 
+### console 2.0
```
Comment (JamesGardiner) [inline CHANGELOG.md]: Can we addmove these to unreleased until we cut a tag.

File: console/views.py [hunk 2]
```diff
@@ -1,15 +1,19 @@
-from flask import request, render_template, jsonify
-
-from console import app
-
 import console.settings as settings
```
Comment (JamesGardiner) [inline console/views.py]: Import order should be PEP8 compliant

File: tests/test_console.py [hunk 3]
```diff
@@ -0,0 +1,21 @@
+from server import app
```
Comment (JamesGardiner) [inline tests/test_console.py]: Import order should be PEP8 compliant

File: console/views.py [hunk 4]
```diff
@@ -1,15 +1,19 @@
...
 
 
 logging.basicConfig(level=settings.LOGGING_LEVEL, format=settings.LOGGING_FORMAT)
 logger = wrap_logger(logging.getLogger(__name__))
+
+
+def heartbeat(logger=logger, heartbeats=0):
+    logger.info("Heartbeat " + str(heartbeats))
+    heartbeats += 1
+    Timer(30.0, heartbeat, [logger, heartbeats]).start()
```
Comment (JamesGardiner) [inline console/views.py]: Remove the hardcoded delay and use an env var instead.

File: CHANGELOG.md (hunk 1 above)
Comment (necrophonic) [inline CHANGELOG.md]: Just leave these in the "unreleased" block for now - they'll get tagged as 2.0.0 when the release is actually taken.

File: console/settings.py [hunk 5]
```diff
@@ -7,57 +7,57 @@
...
-RABBIT_URL2 = 'amqp://{user}:{password}@{hostname}:{port}/{vhost}'.format(
-    hostname=os.getenv('RABBITMQ_HOST2', 'rabbit'),
-    port=os.getenv('RABBITMQ_PORT2', 5672),
-    user=os.getenv('RABBITMQ_DEFAULT_USER', 'rabbit'),
-    password=os.getenv('RABBITMQ_DEFAULT_PASS', 'rabbit'),
-    vhost=os.getenv('RABBITMQ_DEFAULT_VHOST', '%2f')
-)
-
-RABBIT_URLS = [RABBIT_URL, RABBIT_URL2]
+# def get_key(key_name):
```
Comment (necrophonic) [inline console/settings.py]: If code is not needed anymore just delete it (it's in git anyhow) rather than comment out.

File: console/views.py [hunk 6]
```diff
@@ -1,15 +1,19 @@
...
+
+from threading import Timer
 
 
 logging.basicConfig(level=settings.LOGGING_LEVEL, format=settings.LOGGING_FORMAT)
 logger = wrap_logger(logging.getLogger(__name__))
+
+
+def heartbeat(logger=logger, heartbeats=0):
+    logger.info("Heartbeat " + str(heartbeats))
```
Comment (necrophonic) [inline console/views.py]: You've got a `structlog` so put the heartbeats into a key/value pair:
```python
logger.info("Heartbeat", heartbeats=heartbeats)
```

Though do we need the actual number of beats? What's the maximum storable int value in `heartbeats` and what would happen on overflow if we just keep counting up (and the service stays up).

Is there a performance/memory implication for the recursive calls?

File: test_requirements.txt [hunk 7]
```diff
@@ -1,5 +1,7 @@
-mccabe==0.4.0
 flake8==2.5.4
```
Comment (JamesGardiner) [inline test_requirements.txt:1]: Import order should be pep8 compliant

File: tests/test_console.py [hunk 8]
```diff
@@ -0,0 +1,21 @@
...
+        self.app = app.test_client()
+        # propagate the exceptions to the test client
+        self.app.testing = True
+
+    @log_capture()
+    def test_heartbeat(self, l):
+        sleep(30.0)
+        l.check(
+            ('console.views', 'INFO', "event='Heartbeat 1'")
+        )
```
Comment (JamesGardiner) [inline tests/test_console.py]: Move this to server.py and implement the class I sent you - this should make graceful shutdown easier in dev/test/prod environments.

**Review by JamesGardiner — CHANGES_REQUESTED** [review]

**Comment by andrewmil** [comment]
Using a new method for the heartbeat recommended by James. Client now shuts down after tests run. 

I believe the method in server.py stiff runs recursively. Not sure how this will affect performance but this seems to be a standard solution looking around online

File: CHANGELOG.md [hunk 9]
```diff
@@ -7,5 +7,9 @@
   - Add QBS template
   - Add all environment variables to README
```
Comment (JamesGardiner) [inline CHANGELOG.md]: Remove space

File: README.md [hunk 10]
```diff
@@ -42,3 +42,4 @@ Some of important environment variables available for configuration are listed b
 | SDX_FTP_IMAGE_PATH      | `EDC_QImages`                         | FTP Image path
 | SDX_FTP_DATA_PATH       | `EDC_QData`                           | FTP Data path
 | SDX_FTP_RECEIPT_PATH    | `EDC_QReceipts`                       | FTP Receipt path
+| HB_INTERVAL             | '30'                                  | Interval for console heartbeat
```
Comment (JamesGardiner) [inline README.md]: Use backticks to keep formatting consistent

File: console/settings.py [hunk 11]
```diff
@@ -6,58 +6,4 @@
...
-RABBIT_URL2 = 'amqp://{user}:{password}@{hostname}:{port}/{vhost}'.format(
-    hostname=os.getenv('RABBITMQ_HOST2', 'rabbit'),
-    port=os.getenv('RABBITMQ_PORT2', 5672),
-    user=os.getenv('RABBITMQ_DEFAULT_USER', 'rabbit'),
-    password=os.getenv('RABBITMQ_DEFAULT_PASS', 'rabbit'),
-    vhost=os.getenv('RABBITMQ_DEFAULT_VHOST', '%2f')
-)
-
-RABBIT_URLS = [RABBIT_URL, RABBIT_URL2]
+HB_INTERVAL = os.getenv("$HB_INTERVAL", 30)
```
Comment (JamesGardiner) [inline console/settings.py]: The $ can be removed

File: server.py [hunk 12]
```diff
@@ -1,12 +1,50 @@
...
-import logging
 import os
+from threading import Timer
+import logging
+
+from structlog import wrap_logger
 
+from console import app
 from console import __version__
+import console.settings as settings
```
Comment (JamesGardiner) [inline server.py]: Can this be `from console import settings`

File: server.py [hunk 13]
```diff
@@ -1,12 +1,50 @@
...
+
+def heartbeat(logger=logger):
+    logger.info("Heartbeat")
 
-logger = logging.getLogger(__name__)
 
 if __name__ == '__main__':
     logger.info("Starting server: version='{}'".format(__version__))
-    port = int(os.getenv("PORT"))
+    port = int(os.getenv("$PORT", 5000))
```
Comment (JamesGardiner) [inline server.py]: Remove $

**Review by JamesGardiner — CHANGES_REQUESTED** [review]

**Review by JamesGardiner — APPROVED** [review]

**Pull request merged**